### PR TITLE
VACMS 15823 - Provider Core Training icon and wording

### PR DESCRIPTION
**Pull request by eselkin**
## Summary

- If trainings on provider has trainings from PPMS then displays icon and "Provider core training" wording. Only added to `CCProviderResult.jsx`, `UrgentCareResult.jsx`, and `EmergencyCareResult.jsx` result items types.
- This is the solution because we did not display data related to trainings before.
- Sitewide Facilities

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/15823

## Testing done

- Tested manually since training data is still incredibly sparse
- Added automated testing for rendering new component

## Screenshots

<details>
<summary>Shows a provider with and without training</summary>

Screenshot 2024-07-09 at 3 31 56 PM

</details>

## What areas of the site does it impact?

Facility Locator

## Acceptance criteria

- [x] Recommend the label reads "Provider core training" (no "complete"), no link out
- [x] When facility type = "Community providers in VA's network", community providers with training data display the training icon and label.
- [x] When facility type = "Urgent care" and service type = "All in network urgent care" or "In network community urgent care", community providers with training data display the training icon and label.
- [x] When facility type = "Emergency care" and service type = "All in network emergency care" or "In network community emergency care", community providers with training data display the training icon and label.
- [x] Icon and label display consistent with design
- [x] icon uses aria-hidden = true
- [x] Only providers with training data display the icon and label
- [ ] Design review and signoff
- [ ] Accessibility review and signoff
- [ ] Po/PM review and signoff

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user (non authed feature)

## Requested Feedback

Will have to use review instance or local deployment of vets-website (Can't test on tugboat because of facility locator)
1. Go to /find-locations
2. Use address: 2414 Walbert Ave, Allentown, PA 18104
3. Use Community providers for facility type
4. Wait for box on right to allow typing
5. Type in Podiatrist
6. Click Search, look at results

If running local vets-website, you can use `yarn watch --env api=https://dev-api.va.gov` and then go to `http://localhost:3001/find-locations` if you have the `MAPBOX_TOKEN` in your .env

### CODE REVIEW

**Review by laflannery — CHANGES_REQUESTED** [review]
@eselkin We need to make sure that all content we are adding now is added in a way that that text will be resized properly for browser font settings - the typography updates the DST did allows for this sitewide but we need to make sure our individual products do as well. What this means is that `<p>` tags should be the default and `<div>` and `<span>` should be used less often.

For the Provider core training, currently you have everything wrapped in a `<div>` which has no default font size and will not resize according to browser settings. This screenshot is when I set everything to Very Large, notice how the provider name, direction link, etc are large but not "Provider core training":
image

By changing the wrapping tag to `<p>` this has a font size of 1.06rem and will scale accordingly. Now "Provider core training" is large when I change my font settings:
image

**NOTE**: You may notice that some other elements in the listings also don't scale (address, phone descriptions). This I had already noted in a [separate ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18386)

File: yarn.lock [hunk 1]
```diff
@@ -14654,7 +14654,7 @@ minimist-options@4.1.0:
     is-plain-obj "^1.1.0"
     kind-of "^6.0.3"
 
-minimist@1.1.x, minimist@1.2.5, minimist@^1.1.0, minimist@^1.2.0, minimist@^1.2.5, minimist@^1.2.6, minimist@^1.2.8:
+minimist@1.1.x, minimist@1.2.5, minimist@^1.1.0, minimist@^1.2.0, minimist@^1.2.3, minimist@^1.2.5, minimist@^1.2.6, minimist@^1.2.8:
```
Comment (randimays) [inline yarn.lock:14657]: Was this an intentional update?

File: src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { expect } from 'chai';
+import { render } from '@testing-library/react';
+import CCProviderResult from '../../components/search-results-items/CCProviderResult';
+import testData from '../../constants/mock-facility-data-v2.json';
+
+describe('CCProviderResult With Training', () => {
+  it('Should render CCProviderResult, serviceType Podiatrist without training', async () => {
```
Comment (randimays) [inline src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx:8]: Consider adding some line breaks between `it` statements for readability here

File: src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  });
+  it('Should render CCProviderResult, serviceType Podiatrist with 3 training', async () => {
+    const query = {
+      facilityType: 'provider',
+      serviceType: '213E00000X', // Podiatrist
+    };
+    const wrapper = render(
+      <CCProviderResult provider={testData.data[2]} query={query} />,
+    );
+    expect(wrapper.queryAllByTestId('training-c')).to.have.length(1); // stll one
```
Comment (randimays) [inline src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx:38]: Should the comment say `still one`?

File: src/applications/facility-locator/components/search-results-items/common/ProviderTraining.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    return null;
+  }
+  return (
+    <div className="vads-u-margin-y--2" data-testid={`training-${provider.id}`}>
+      <VaIcon
+        size={3}
+        icon="how_to_reg"
+        className="vads-u-margin-right--1 success-icon"
+      />
+      <span>Provider core training</span>
```
Comment (randimays) [inline src/applications/facility-locator/components/search-results-items/common/ProviderTraining.jsx:16]: +1 on Laura's comment about using semantic markup here

**Review by randimays — CHANGES_REQUESTED** [review]
Left a few minor comments

File: src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx (hunk 3 above)
Comment (eselkin) [reply src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx:38]: Should probably say still 1 with 3 trainings.

File: yarn.lock (hunk 1 above)
Comment (eselkin) [reply yarn.lock:14657]: Nope, thanks

File: src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx (hunk 2 above)
Comment (eselkin) [reply src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx:8]: added

File: src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx (hunk 3 above)
Comment (eselkin) [reply src/applications/facility-locator/tests/components/ProviderTraining.unit.spec.jsx]: changed

File: src/applications/facility-locator/components/search-results-items/common/ProviderTraining.jsx (hunk 4 above)
Comment (eselkin) [reply src/applications/facility-locator/components/search-results-items/common/ProviderTraining.jsx]: changed

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Review by laflannery — APPROVED** [review]
Looks good!

**Review by thejordanwood — APPROVED** [review]
Approving based on the screenshots.

**Review by randimays — APPROVED** [review]
LGTM

**Pull request merged**